subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy

```diff
@@ -588,4 +588,197 @@ class ValidatePluginsIntegrationTest extends AbstractPluginValidationIntegration
         "InputFile"  | _
         "InputFiles" | _
     }
+    @ValidationTestFor([
+            ValidationProblemId.MISSING_NORMALIZATION_ANNOTATION,
+            ValidationProblemId.INCORRECT_USE_OF_INPUT_ANNOTATION
+    ])
+    def "detects problems with file inputs"() {
+        file("input.txt").text = "input"
+        file("input").createDir()
+
+        javaTaskSource << """
+            import org.gradle.api.*;
+            import org.gradle.api.file.*;
+            import org.gradle.api.tasks.*;
+            import java.util.Set;
+            import java.util.Collections;
+            import java.io.File;
+            import java.nio.file.Path;
+
+            @CacheableTask
+            public abstract class MyTask extends DefaultTask {
```

Comment:
If you use the same task type in both tests, maybe it would make sense to extract this into a helper method.

Reply:
I'll do this.

Reply:
The Task type is actually very different in both tests.